app.ts

```diff
@@ -538,10 +538,16 @@ async function main() {
     if (opts.prediscovered) {
         const prediscoveredCompilersJson = await fs.readFile(opts.prediscovered, 'utf8');
         initialCompilers = JSON.parse(prediscoveredCompilersJson);
-        await compilerFinder.loadPrediscovered(initialCompilers);
+        const prediscResult = await compilerFinder.loadPrediscovered(initialCompilers);
+        if (prediscResult.length === 0) {
```

Comment:
these are extra checks, but empty initialCompilers will not cause compilation errors if you call the /compile endpoint, but it does cause en empty UI without languages and compilers which is not very user friendly